parsers/ocaml.c

```diff
@@ -873,23 +920,33 @@ static void killCurrentState ( void )
 		/* nothing more */
 		break;
 	}
+	return popped;
 }
 
-/* used to prepare tag for OCaml, just in case their is a need to
+/* Keep track of our _true_ line number and file pos,
+ * as the lookahead token gives us false values. */
+static unsigned long ocaLineNumber;
+static fpos_t ocaFilePosition;
+
+/* used to prepare an OCaml tag , just in case their is a need to
  * add additional information to the tag. */
 static void prepareTag (tagEntryInfo * tag, vString const *name, ocamlKind kind)
 {
 	int parentIndex;
 
-	initTagEntry (tag, vStringValue (name));
+	/* Ripped out of read.h initTagEntry, because of line number
+	 * shenanigans.
```

Comment:
I'd rather keep the `initTagEntry()` call and only override the required fields,  or use the new `initTagEntryFull()`.
This can help if we need to add new fields in the future.